### PR TITLE
Guess Who? game project Irina

### PR DESCRIPTION
**Pull request by Irina-web-dev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+
+At some point, the user wants to make a guess about who the secret person is. The cards have a "Guess" button which is connected to the guess function via the onclick attribute on the card element. The guess function contains confirm() method to make sure that the user really wants to make a guess, as this will end the game in one way or another. If the player confirms that they want to make a guess, the checkMyGuess function is invoked. The chechMyGuess function contains if statement to check if the suspect is the same as the secret person's name and to set a message to show in the win or lose section accordingly.
+
+Finally, the event listener is added to restart button that invokes the start function if the user wants to start the game over again.
+
+If I had more time I would have created a timer that measures how long time each game takes and a counter to keep track of how many guesses a player does.
```
Comment (dannuzak) [inline README.md:19]: Great description of the project! it really shows your good understanding of each step! I wonder if you found challenges and how you solved them.

File: code/script.js [hunk 2]
```diff
@@ -303,41 +331,92 @@ const filterCharacters = (keep) => {
...
-  } else if (group === 'other') {
-    // Similar to the one above
+  } else if (category === 'other') {
+    if (keep) {
+      alert(
+        `Yes, the person is ${attribute}! Keep all persons who are ${attribute}`
+      )
+    } else {
+      alert(
+        `No, the person is not ${attribute}! Remove all persons who are not ${attribute}`
```
Comment (dannuzak) [inline code/script.js:341]: A little observation while playing the game. Here when the secret person is "not smoking" the program removes the persons who are smoking ( as it is meant to do) but the message is saying that will remove the persons who are "NOT smoking". Maybe the word "not" in the alert should be removed so it matches the action of the program?

File: code/script.js [hunk 3]
```diff
@@ -253,47 +256,72 @@ const setSecret = () => {
...
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
+  let keep;
+  let valueSecret = secret[currentQuestion.attribute];
+
+  if(valueSecret === currentQuestion.value){
+    keep = true
+  } else {
+    keep = false
+  }
```
Comment (dannuzak) [inline code/script.js:312]: Well done! I learned from your approach! :)

File: code/script.js [hunk 4]
```diff
@@ -303,41 +331,92 @@ const filterCharacters = (keep) => {
...
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter(person => person[currentQuestion.attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter(person => person[currentQuestion.attribute] !== value)
+  }
```
Comment (dannuzak) [inline code/script.js:371]: Well done!

File: code/script.js [hunk 5]
```diff
@@ -303,41 +331,92 @@ const filterCharacters = (keep) => {
...
+findOutButton.addEventListener('click', checkQuestion)
+// Event listener which occurs when the user clicks on the play again button
+playAgainButton.addEventListener('click', () => {
+  //Hide the win or lose section
+  winOrLose.style.display = "none";
+  //Show the board again
+  board.style.display = "flex";
+  //Invoke start function to start a new game
+  start() 
+})
```
Comment (dannuzak) [inline code/script.js:422]: Looks great!

File: README.md [hunk 6]
```diff
@@ -1,13 +1,23 @@
 # Project Name
 
-Replace this readme with your own information about your project.
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+The task was to create digital version of the character guessing board game Guess Who? by focusing mainly on using JavaScript.
```
Comment (dannuzak) [inline README.md:3]: I really enjoyed reading your code Irina! Clean and effective, keep up the great job! 👍🏼  😀  👯‍♀️

File: code/script.js [hunk 7]
```diff
@@ -253,47 +256,72 @@ const setSecret = () => {
...
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-
+  const category = questions.options[questions.selectedIndex].parentNode.label
+  // This variable stores the actual value of the question we've selected.
+  actualValue = questions.value
```
Comment (KarolCislowski) [inline code/script.js:270]: You do not use this variable outside this function, so it shouldn't be created in the global scope, but here. 🤓

**Review by KarolCislowski — APPROVED** [review]
This is a very nice project 🌈 

I know it was an extra point of this assignment to style it your way but remember that there is no another way to master CSS then practice it over and over again 🦾 

Remember that you can always come back to it when you have some spare time and add more functionalities to make it an even greater portfolio asset! 🦄

**Pull request closed without merging**